rotkehlchen/globaldb/updates.py

```diff
@@ -300,74 +306,230 @@ def _parse_full_insert(self, insert_text: str) -> AssetData:
-                if local_asset is None:
-                    try:  # if asset is not known then simply do an insertion
-                        with connection.savepoint_ctx() as cursor:
-                            executeall(cursor, full_insert)
-                    except sqlite3.Error as e:
-                        self.msg_aggregator.add_warning(
-                            f'Failed to add asset {remote_asset_data.identifier} in the '
-                            f'DB during the v{version} assets update. Skipping entry. '
-                            f'Error: {str(e)}',
-                        )
-                    continue  # fail or succeed continue to next entry
-
-                # otherwise asset is known, so it's a conflict. Check if we can resolve
-                resolution = conflicts.get(local_asset) if conflicts else None
-                if resolution == 'local':
-                    # do nothing, keep local
+                    AssetResolver().clean_memory_cache(identifier=local_asset.identifier)
+        except sqlite3.Error:  # https://docs.python.org/3/library/sqlite3.html#exceptions
+            if local_asset is None:
+                try:  # if asset is not known then simply do an insertion
+                    with connection.savepoint_ctx() as cursor:
+                        executeall(cursor, full_insert)
+                except sqlite3.Error as e:
+                    self.msg_aggregator.add_warning(
+                        f'Failed to add asset {remote_asset_data.identifier} in the '
+                        f'DB during the v{version} assets update. Skipping entry. '
+                        f'Error: {str(e)}',
+                    )
+                return  # fail or succeed continue to next entry
+
+            if local_asset is not None:
+                AssetResolver().clean_memory_cache(local_asset.identifier.lower())
+
+            # otherwise asset is known, so it's a conflict. Check if we can resolve
+            resolution = assets_conflicts.get(local_asset) if assets_conflicts else None
+            if resolution == 'local':
+                # do nothing, keep local
+                return
+            if resolution == 'remote':
+                try:
+                    with connection.savepoint_ctx() as cursor:
+                        _force_remote_asset(cursor, local_asset, full_insert)
+                except sqlite3.Error as e:
+                    self.msg_aggregator.add_warning(
+                        f'Failed to resolve conflict for {remote_asset_data.identifier} in '
+                        f'the DB during the v{version} assets update. Skipping entry. '
+                        f'Error: {str(e)}',
+                    )
+                return  # fail or succeed continue to next entry
+
+            # else can't resolve. Mark it for the user to resolve.
+            # TODO: When assets refactor is finished, remove the usage of AssetData here
```

Comment:
question: This todo still relevant? And I think I have asked that same question before. What do we mean by "assets refactor" here?

Reply:
Discussed in discord. This will come in a future PR